### ISSUE TITLE
Mix module/text and script

### ISSUE THREAD

**Issue by Isotop7** [jaagr/polybar#331]
Is there a possibilty to 'combine' the modules text and script to set the modules content 

- to the output of a script
- and not invoke it in an interval

?

This would be useful to let scripts echo a message and polybar would draw it as the texts content.

If this is already possible, please ELI5 and close this issue.

**Comment by jaagr** [comment]
Have you tried the ipc module? It's handing when used together with `polybar-msg`. It allows you to define a set of hooks that can be triggered using:

~~~ sh
$ polybar-msg hook <module-name> <index>
~~~

**Comment by Isotop7** [comment]
My purpose would be:

Run script A on click of module
Script echos a message
Set the content of the module to the echoed message

Would this be possible?

**Comment by jaagr** [comment]
Yeah but you could use a regular script module for that, for example:

~~~ dosini
[module/demo]
type = custom/script
exec = mkfifo /tmp/pipe.demo && echo initial output >/tmp/pipe.demo & tail -f /tmp/pipe.demo
tail = true
click-left = date +%s >/tmp/pipe.demo
~~~

**Comment by jaagr** [comment]
And here's an example with the ipc module, which adds some more flexibility:

~~~ dosini
[module/demo]
type = custom/ipc
hook-0 = echo foobar
hook-1 = date +%s
hook-2 = whoami
initial = 1
click-left = polybar-msg hook demo 1
click-right = polybar-msg hook demo 2
double-click-left = polybar-msg hook demo 3
~~~

**Comment by Isotop7** [comment]
@jaagr ty. ipc works flawlessly :+1:
